Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -224,12 +261,140 @@ void webkit_user_content_manager_remove_all_scripts(WebKitUserContentManager* ma
     manager->priv->userContentController->removeAllUserScripts();
 }
 
+/**
+ * WebKitScriptMessageReply: (ref-func webkit_script_message_reply_ref) (unref-func webkit_script_message_reply_unref)
+ *
+ * A reply for a script message received.
+ * If no reply has been sent by the user, an automatically generated reply with
+ * undefined value with be sent.
+ *
+ * Since: 2.40
+ */
+struct _WebKitScriptMessageReply {
+    _WebKitScriptMessageReply(WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler)
+        : completionHandler(WTFMove(completionHandler))
+        , referenceCount(1)
+    {
+    }
+
+    void sendValue(JSCValue* value)
+    {
+        JSCContext* jscContext = SharedJavascriptContext::singleton().getOrCreateContext();
```

Comment:
After commit https://github.com/WebKit/WebKit/commit/4041ec7696c53b8f7e64d3046d4f7ae3d180cf35 you can now change this to be:

```suggestion
        JSCContext* jscContext = API::SerializedScriptValue::sharedJSCContext();
```

Reply:
Thank you for the suggestions! A dumb question: do I have to include the 4041ec7 commit in the forked repo for pull request in order to pass the build test?

Reply:
Yeah, you'll want to rebase.

Reply:
Encountering a problem here: after replacing `SharedJavascriptContext::singleton().getOrCreateContext()` to `API::SerializedScriptValue::sharedJSCContext()` in Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp, ld.lld complains about "undefined symbol", when "APISerializedScriptValue.h" is already included. 
May I ask what's going wrong here?

Reply:
Looks like it is not exported... try this (untested):

```
diff --git a/Source/WebKit/Shared/API/APISerializedScriptValue.h b/Source/WebKit/Shared/API/APISerializedScriptValue.h
index 8fb9ba26ecac..ddb67f89af1e 100644
--- a/Source/WebKit/Shared/API/APISerializedScriptValue.h
+++ b/Source/WebKit/Shared/API/APISerializedScriptValue.h
@@ -73,7 +73,7 @@ public:
 #endif
 
 #if USE(GLIB)
-    static JSCContext* sharedJSCContext();
+    WK_EXPORT static JSCContext* sharedJSCContext();
     static GRefPtr<JSCValue> deserialize(WebCore::SerializedScriptValue&);
     static RefPtr<SerializedScriptValue> createFromGVariant(GVariant*);
 #endif
 ```

Reply:
Thank you! Now it's working

Reply:
After this replacement of shared context, though, the script message handler is failing the unit test locally; I'm not sure why this is the case?

Reply:
Also I'm not quite certain about the reason why previously the passed tests in a local environment were unable to pass the EWS bot's APITests... Is there some sort of caching mechanism at work here that tricked me into believing the changed code was working? Any input would be appreciated 🙏